### ISSUE TITLE
Error with untitled document on hover

### ISSUE THREAD

**Issue by MoonE** [vimeo/psalm#9450]
It's no big deal to save the document, but maybe this could be handled more gracefully?

> [INFO  - 1:13:58 AM] Psalm Version was detected as 5.7.7
> [Error - 1:20:25 AM] Request textDocument/hover failed.
>   Message: InvalidArgumentException: Not a valid file URI: untitled:Untitled-4 in /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php:532
> Stack trace:
> #&#8203;0 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/Server/TextDocument.php(205): Psalm\Internal\LanguageServer\LanguageServer::uriToPath()
> #&#8203;1 /phpmyadmin/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(168): Psalm\Internal\LanguageServer\Server\TextDocument->hover()
> #&#8203;2 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(144): AdvancedJsonRpc\Dispatcher->dispatch()
> #&#8203;3 [internal function]: Psalm\Internal\LanguageServer\LanguageServer->Psalm\Internal\LanguageServer\{closure}()
> #&#8203;4 /phpmyadmin/vendor/amphp/amp/lib/Coroutine.php(67): Generator->current()
> #&#8203;5 /phpmyadmin/vendor/amphp/amp/lib/functions.php(96): Amp\Coroutine->__construct()
> #&#8203;6 /phpmyadmin/vendor/amphp/amp/lib/functions.php(61): Amp\call()
> #&#8203;7 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/EmitterTrait.php(85): Amp\{closure}()
> #&#8203;8 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/ProtocolStreamReader.php(114): Psalm\Internal\LanguageServer\ProtocolStreamReader->emit()
> #&#8203;9 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/ProtocolStreamReader.php(63): Psalm\Internal\LanguageServer\ProtocolStreamReader->readMessages()
> #&#8203;10 [internal function]: Psalm\Internal\LanguageServer\ProtocolStreamReader->Psalm\Internal\LanguageServer\{closure}()
> #&#8203;11 /phpmyadmin/vendor/amphp/amp/lib/Coroutine.php(118): Generator->send()
> #&#8203;12 /phpmyadmin/vendor/amphp/amp/lib/Internal/Placeholder.php(149): Amp\Coroutine->Amp\{closure}()
> #&#8203;13 /phpmyadmin/vendor/amphp/amp/lib/Deferred.php(53): Amp\Promise@anonymous->resolve()
> #&#8203;14 /phpmyadmin/vendor/amphp/byte-stream/lib/ResourceInputStream.php(101): Amp\Deferred->resolve()
> #&#8203;15 /phpmyadmin/vendor/amphp/amp/lib/Loop/NativeDriver.php(327): Amp\ByteStream\ResourceInputStream::Amp\ByteStream\{closure}()
> #&#8203;16 /phpmyadmin/vendor/amphp/amp/lib/Loop/NativeDriver.php(127): Amp\Loop\NativeDriver->selectStreams()
> #&#8203;17 /phpmyadmin/vendor/amphp/amp/lib/Loop/Driver.php(138): Amp\Loop\NativeDriver->dispatch()
> #&#8203;18 /phpmyadmin/vendor/amphp/amp/lib/Loop/Driver.php(72): Amp\Loop\Driver->tick()
> #&#8203;19 /phpmyadmin/vendor/amphp/amp/lib/Loop.php(95): Amp\Loop\Driver->run()
> #&#8203;20 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(516): Amp\Loop::run()
> #&#8203;21 /phpmyadmin/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(331): Psalm\Internal\Analyzer\ProjectAnalyzer->server()
> #&#8203;22 /phpmyadmin/vendor/vimeo/psalm/psalm-language-server(9): Psalm\Internal\Cli\LanguageServer::run()
> #&#8203;23 {main}
>   Code: -32603

**Comment by orklah** [comment]
Hey @tm1000! Seems like something that could be improved in LanguageServer

**Comment by tm1000** [comment]
@MoonE whats the context of this situation. Seems like a file that doesn't exist on the disk?

**Comment by MoonE** [comment]
@tm1000 Yeah, I use VSCode with the psalm-vscode-plugin. Made a new file with a php class. This error message is printed when hovering over any code with the mouse pointer.

**Comment by tm1000** [comment]
@orklah @weirdan  assign to me please

**Comment by orklah** [comment]
Done!
